tests/basic_test.cpp

```diff
@@ -373,6 +373,42 @@ BOOST_AUTO_TEST_CASE(exception_in_startup)
    app_thread.join();
 }
 
+// -----------------------------------------------------------------------------
+// Here we make sure that if a plugin calls app().quit() during `plugin_startup()`,
+// it doesn't interrupt the startup process for other plugins
+// `producer_plugin` can do this and some tests like `terminate-scenarios-test-hard_replay`
+// rely on other plugins getting initialized.
+// -----------------------------------------------------------------------------
+BOOST_AUTO_TEST_CASE(quit_in_startup)
+{
+   appbase::application::register_plugin<pluginB>();
+
+   appbase::scoped_app app;
+
+   const char* argv[] = { bu::framework::current_test_case().p_name->c_str(),
+                          "--plugin", "pluginA", "--log", "--quit_during_startup",
+                          "--plugin", "pluginB", "--log2" };
+
+   BOOST_CHECK(app->initialize<pluginB>(sizeof(argv) / sizeof(char*), const_cast<char**>(argv)));
+
+   std::thread app_thread( [&]() {
+      auto& pA = app->get_plugin<pluginA>();
```

Comment:
A minor thing. Our code uses `p_A/plugin_A` style instead of `pA/pluginA`. You don't have to change now.